toolchain/semantics/semantics_handle.cpp

```diff
@@ -233,14 +237,23 @@ auto SemanticsHandleLiteral(SemanticsContext& context,
            .exponent = token_value.Exponent(),
            .is_decimal = token_value.IsDecimal()});
       context.AddNodeAndPush(parse_node,
-                             SemanticsNode::RealLiteral::Make(parse_node, id));
+                             SemanticsNode::RealLiteral::Make(
+                                 parse_node,
+                                 context.CanonicalizeType(
+                                     SemanticsNodeId::BuiltinFloatingPointType),
+                                 id));
       break;
     }
     case TokenKind::StringLiteral: {
       auto id = context.semantics().AddString(
           context.tokens().GetStringLiteral(token));
+      context.CanonicalizeType(SemanticsNodeId::BuiltinStringType);
```

Comment:
A bit confused why this canonicalizes the builtin string type twice? Should this be removed in favor of the code below, or use a variable?

Reply:
Cleaning up, this is a mistake.